From 0a24a0566aa7f2a1905b4f2d5f9fa54ef63e5396 Mon Sep 17 00:00:00 2001
From: tenzap <fabstz-it@yahoo.fr>
Date: Tue, 16 Nov 2021 18:08:33 +0100
Subject: [PATCH] Fix compilation issue on older mac (< 10.12)

Error was:
In file included from src/lib/entropy/getentropy/getentropy.cpp:13:
/usr/include/sys/random.h:37:32: error: unknown type name 'u_int'
void read_random(void* buffer, u_int numBytes);
                               ^
/usr/include/sys/random.h:38:33: error: unknown type name 'u_int'
void read_frandom(void* buffer, u_int numBytes);
                                ^
/usr/include/sys/random.h:39:33: error: unknown type name 'u_int'
int  write_random(void* buffer, u_int numBytes);
                                ^
--- src/lib/entropy/getentropy/getentropy.cpp
+++ src/lib/entropy/getentropy/getentropy.cpp
@@ -10,6 +10,9 @@
 #if defined(BOTAN_TARGET_OS_IS_OPENBSD) || defined(BOTAN_TARGET_OS_IS_FREEBSD)
    #include <unistd.h>
 #else
+   #if defined(BOTAN_TARGET_OS_HAS_POSIX1)
+      #include <sys/types.h>
+   #endif
    #include <sys/random.h>
 #endif
 
